packages/system/src/Item/Components/StockItemSelectModal/StockItemSelectModal.tsx

```diff
@@ -70,24 +73,44 @@ export const StockItemSelectModal = ({
 
   const onChangeItems: AutocompleteOnChange<
     ItemRowWithStatsFragment | ItemRowWithStatsFragment[]
-  > = (_event, items) => {
-    if (items && items instanceof Array) {
-      setItemIds(items.map(item => item.id));
+  > = (_event, items) => setSelectedItems(items instanceof Array ? items : []);
+
+  const selectAll = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const filtered = options.filter(option =>
+      RegexUtils.matchNameOrCode(option, inputValue)
+    );
+    if (event.target.checked) {
+      setSelectedItems([...selectedItems, ...filtered]);
+    } else {
+      const filteredIds = filtered.map(item => item.id);
+      setSelectedItems(
+        selectedItems.filter(item => !filteredIds.includes(item.id))
+      );
     }
   };
 
   const ItemInput: FC<AutocompleteRenderInputParams> = props => {
     const { InputProps, ...rest } = props;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { startAdornment, ...restInputProps } = InputProps ?? {};
     const t = useTranslation('common');
-    const length =
-      startAdornment && startAdornment instanceof Array
-        ? startAdornment.length
-        : 0;
 
     return (
       <>
-        <Typography>{t('label.items-selected', { count: length })}</Typography>
+        <Box display="flex">
+          <Typography
+            flex={1}
+            style={{ verticalAlign: 'bottom' }}
+            display="flex"
+            alignItems="center"
+          >
+            {t('label.items-selected', { count: selectedItems.length })}
+          </Typography>
+          <Typography textAlign="right" flex={1}>
+            {t('label.select-all')}
+            <Checkbox onChange={selectAll} />
```

Comment:
Would be nice if the checkbox value showed the "partial" state in response to selecting/un-selecting individual items, like the Tables do:
<img width="220" alt="Screen Shot 2022-03-25 at 2 22 54 PM" src="https://user-images.githubusercontent.com/5456533/160037375-8f504bbe-fd8a-40a3-9487-6bc6326e8b44.png">

Although... I'm not sure what the logic would be when there is a filter applied -- should the "Select all" checkbox state reflect *all* the items or just the currently filtered ones? I want to say *all* the items, but then when you *click* "Select All" it only updates the *filtered* items. And that's kind of confusing so maybe we should just leave it as it is now. Thoughts?

Reply:
The intention was to make it possible to select a filtered list of items, similar to what you can do now in mSupply.
For that - the 'select all' action has to operate on the filtered / displayed list of items. In which case it would make sense that the checked state reflected the visible list. Just not sure if it's possible. Will see.

Reply:
turns out, not that hard. and much nicer to use 😁 
thanks!
97926b7b